<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Harmonise the alleles and effects between the exposure and outcome — harmonise_data • TwoSampleMR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Harmonise the alleles and effects between the exposure and outcome — harmonise_data"><meta property="og:description" content="In order to perform MR the effect of a SNP on an outcome and exposure must be harmonised to be
relative to the same allele."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mrcieu/TwoSampleMR" class="external-link">Source</a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Harmonise the alleles and effects between the exposure and outcome</h1>
    
    <div class="hidden name"><code>harmonise_data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>In order to perform MR the effect of a SNP on an outcome and exposure must be harmonised to be
relative to the same allele.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">harmonise_data</span><span class="op">(</span><span class="va">exposure_dat</span>, <span class="va">outcome_dat</span>, action <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>exposure_dat</dt>
<dd><p>Output from <code>read_exposure_data</code>.</p></dd>


<dt>outcome_dat</dt>
<dd><p>Output from <code>extract_outcome_data</code>.</p></dd>


<dt>action</dt>
<dd><p>Level of strictness in dealing with SNPs.</p><ul><li><p><code>action = 1</code>: Assume all alleles are coded on the forward strand, i.e. do not attempt to flip alleles</p></li>
<li><p><code>action = 2</code>: Try to infer positive strand alleles, using allele frequencies for palindromes (default, conservative);</p></li>
<li><p><code>action = 3</code>: Correct strand for non-palindromic SNPs, and drop all palindromic SNPs from the analysis (more conservative).
If a single value is passed then this action is applied to all outcomes.
But multiple values can be supplied as a vector, each element relating to a different outcome.</p></li>
</ul></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Data frame with harmonised effects and alleles</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Expects data in the format generated by <code>read_exposure_data</code> and <code>extract_outcome_data</code>.
This means the inputs must be dataframes with the following columns:</p>
<p><code>outcome_dat</code>:</p><ul><li><p><code>SNP</code></p></li>
<li><p><code>beta.outcome</code></p></li>
<li><p><code>se.outcome</code></p></li>
<li><p><code>effect_allele.outcome</code></p></li>
<li><p><code>other_allele.outcome</code></p></li>
<li><p><code>eaf.outcome</code></p></li>
<li><p><code>outcome</code></p></li>
</ul><p><code>exposure_dat</code>:</p><ul><li><p><code>SNP</code></p></li>
<li><p><code>beta.exposure</code></p></li>
<li><p><code>se.exposure</code></p></li>
<li><p><code>effect_allele.exposure</code></p></li>
<li><p><code>other_allele.exposure</code></p></li>
<li><p><code>eaf.exposure</code></p></li>
</ul><p>The function tries to harmonise INDELs. If they are coded as sequence strings things work more smoothly.
If they are coded as D/I in one dataset it will try to convert them to sequences if the other dataset has adequate information.
If coded as D/I in one dataset and as a variant with equal length INDEL alleles in the other, the variant is dropped.
If one or both the datasets only has one allele (i.e. the effect allele) then harmonisation is naturally going to be more ambiguous and more variants will be dropped.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

