<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Read exposure data — read_exposure_data • TwoSampleMR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Read exposure data — read_exposure_data"><meta property="og:description" content="Reads in exposure data. Checks and organises columns for use with MR or enrichment tests. Infers p-values when possible from beta and se. Looks up SNPs in biomaRt to get basic info."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/mrcieu/TwoSampleMR" class="external-link">Source</a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read exposure data</h1>
    
    <div class="hidden name"><code>read_exposure_data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Reads in exposure data. Checks and organises columns for use with MR or enrichment tests. Infers p-values when possible from beta and se. Looks up SNPs in biomaRt to get basic info.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">read_exposure_data</span><span class="op">(</span></span>
<span>  <span class="va">filename</span>,</span>
<span>  clump <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">" "</span>,</span>
<span>  phenotype_col <span class="op">=</span> <span class="st">"Phenotype"</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  beta_col <span class="op">=</span> <span class="st">"beta"</span>,</span>
<span>  se_col <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>  eaf_col <span class="op">=</span> <span class="st">"eaf"</span>,</span>
<span>  effect_allele_col <span class="op">=</span> <span class="st">"effect_allele"</span>,</span>
<span>  other_allele_col <span class="op">=</span> <span class="st">"other_allele"</span>,</span>
<span>  pval_col <span class="op">=</span> <span class="st">"pval"</span>,</span>
<span>  units_col <span class="op">=</span> <span class="st">"units"</span>,</span>
<span>  ncase_col <span class="op">=</span> <span class="st">"ncase"</span>,</span>
<span>  ncontrol_col <span class="op">=</span> <span class="st">"ncontrol"</span>,</span>
<span>  samplesize_col <span class="op">=</span> <span class="st">"samplesize"</span>,</span>
<span>  gene_col <span class="op">=</span> <span class="st">"gene"</span>,</span>
<span>  id_col <span class="op">=</span> <span class="st">"id"</span>,</span>
<span>  min_pval <span class="op">=</span> <span class="fl">1e-200</span>,</span>
<span>  log_pval <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"chr"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"pos"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>filename</dt>
<dd><p>Filename. Must have header with at least SNP column present.</p></dd>


<dt>clump</dt>
<dd><p>Whether to perform LD clumping with <code><a href="clump_data.html">clump_data</a></code> on the exposure data. The default is <code>FALSE</code>.</p></dd>


<dt>sep</dt>
<dd><p>Specify delimeter in file. The default is a space, i.e. <code>" "</code>.</p></dd>


<dt>phenotype_col</dt>
<dd><p>Optional column name for the column with phenotype name corresponding the the SNP. If not present then will be created with the value "Outcome". The default is <code>"Phenotype"</code>.</p></dd>


<dt>snp_col</dt>
<dd><p>Required name of column with SNP rs IDs. The default is <code>"SNP"</code>.</p></dd>


<dt>beta_col</dt>
<dd><p>Required for MR. Name of column with effect sizes. The default is <code>"beta"</code>.</p></dd>


<dt>se_col</dt>
<dd><p>Required for MR. Name of column with standard errors. The default is <code>"se"</code>.</p></dd>


<dt>eaf_col</dt>
<dd><p>Required for MR. Name of column with effect allele frequency. The default is <code>"eaf"</code>.</p></dd>


<dt>effect_allele_col</dt>
<dd><p>Required for MR. Name of column with effect allele. Must be "A", "C", "T" or "G". The default is <code>"effect_allele"</code>.</p></dd>


<dt>other_allele_col</dt>
<dd><p>Required for MR. Name of column with non effect allele. Must be "A", "C", "T" or "G". The default is <code>"other_allele"</code>.</p></dd>


<dt>pval_col</dt>
<dd><p>Required for enrichment tests. Name of column with p-value. The default is <code>"pval"</code>.</p></dd>


<dt>units_col</dt>
<dd><p>Optional column name for units. The default is <code>"units"</code>.</p></dd>


<dt>ncase_col</dt>
<dd><p>Optional column name for number of cases. The default is <code>"ncase"</code>.</p></dd>


<dt>ncontrol_col</dt>
<dd><p>Optional column name for number of controls. The default is <code>"ncontrol"</code>.</p></dd>


<dt>samplesize_col</dt>
<dd><p>Optional column name for sample size. The default is <code>"samplesize"</code>.</p></dd>


<dt>gene_col</dt>
<dd><p>Optional column name for gene name. The default is <code>"gene"</code>.</p></dd>


<dt>id_col</dt>
<dd><p>Optional column name to give the dataset an ID. Will be generated automatically if not provided for every trait / unit combination. The default is <code>"id"</code>.</p></dd>


<dt>min_pval</dt>
<dd><p>Minimum allowed p-value. The default is <code>1e-200</code>.</p></dd>


<dt>log_pval</dt>
<dd><p>The p-value is -log10(P). The default is <code>FALSE</code>.</p></dd>


<dt>chr_col</dt>
<dd><p>Optional column name for chromosome. Default is <code>"chr"</code>.</p></dd>


<dt>pos_col</dt>
<dd><p>Optional column name for genetic position Default is <code>"pos"</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>data frame</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer></div>

  


  

  </body></html>

