<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction • TwoSampleMR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="TwoSampleMR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mrcieu/TwoSampleMR" class="external-link">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction</h1>
                        <h4 data-toc-skip class="author">Gibran Hemani</h4>
            
      
      
      <div class="hidden name"><code>introduction.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>Two sample Mendelian randomisation (2SMR) is a method to estimate the causal effect of an exposure on an outcome using only summary statistics from genome wide association studies (GWAS). Though conceptually straightforward, there are a number of steps that are required to perform the analysis properly, and they can be cumbersome. The TwoSampleMR package aims to make this easy by combining three important components</p>
<ul>
<li>data management and harmonisation</li>
<li>the statistical routines to estimate the causal effects</li>
<li>connection to a large repository of the actual GWAS summary statistics needed to perform the analyses.</li>
</ul>
<p>The general principles <span class="citation">(G. Davey Smith and Ebrahim 2003; George Davey Smith and Hemani 2014)</span>, and statistical methods <span class="citation">(Pierce and Burgess 2013; Bowden, Davey Smith, and Burgess 2015)</span> can be found elsewhere, here we will just outline how to use the R package.</p>
<p>This package uses the <a href="https://github.com/mrcieu/ieugwasr" class="external-link">ieugwasr</a> package to connect to the database of thousands of complete GWAS summary data.</p>
<hr>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install directly from the GitHub repository do the following:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">"MRCIEU/TwoSampleMR"</span><span class="op">)</span></span></code></pre></div>
<p>If you don’t have the <code>remotes</code> package install it from CRAN using <code>install.packages("remotes")</code>.</p>
<hr>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The workflow for performing MR is as follows:</p>
<ol style="list-style-type: decimal">
<li>Select instruments for the exposure (perform LD clumping if necessary)</li>
<li>Extract the instruments from the <a href="https://gwas.mrcieu.ac.uk/" class="external-link">IEU GWAS database</a> for the outcomes of interest</li>
<li>Harmonise the effect sizes for the instruments on the exposures and the outcomes to be each for the same reference allele</li>
<li>Perform MR analysis, sensitivity analyses, create plots, compile reports</li>
</ol>
<p>A diagramatic overview is shown here:</p>
<p><img src="https://drive.google.com/uc?id=1yHTkn-kVqZzsQfQdXBHZBGdFWj8gSsPQ" width="800"></p>
<p>A basic analysis, e.g. the causal effect of body mass index on coronary heart disease, looks like this:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TwoSampleMR</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># List available GWASs</span></span>
<span><span class="va">ao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_outcomes.html">available_outcomes</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get instruments</span></span>
<span><span class="va">exposure_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_instruments.html">extract_instruments</a></span><span class="op">(</span><span class="st">"ieu-a-2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get effects of instruments on outcome</span></span>
<span><span class="va">outcome_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span><span class="op">(</span>snps<span class="op">=</span><span class="va">exposure_dat</span><span class="op">$</span><span class="va">SNP</span>, outcomes <span class="op">=</span> <span class="st">"ieu-a-7"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Harmonise the exposure and outcome data</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/harmonise_data.html">harmonise_data</a></span><span class="op">(</span><span class="va">exposure_dat</span>, <span class="va">outcome_dat</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform MR</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mr.html">mr</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></code></pre></div>
<p>Each step is documented on other pages in the documentation.</p>
</div>
<div class="section level2">
<h2 id="authentication">Authentication<a class="anchor" aria-label="anchor" href="#authentication"></a>
</h2>
<p><strong>The authentication method has changed recently due to the GoogleAuthR method changing.</strong> The main differences are that:</p>
<ol style="list-style-type: decimal">
<li>By default you will not be asked to authenticate and will only have access to public data</li>
<li>If you do need to authenticate in order to access private datasets there is no longer a single file called <code>mrbase.oauth</code>, rather, there is a directory called <code>ieugwasr_oauth</code>.</li>
</ol>
<p>Detailed information is given here: <a href="https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication" class="external-link">https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication</a></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p><br></p>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bowden2015" class="csl-entry">
Bowden, Jack, George Davey Smith, and Stephen Burgess. 2015. <span>“<span class="nocase">Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression</span>.”</span> <em>International Journal of Epidemiology</em> In press.
</div>
<div id="ref-DaveySmith2003" class="csl-entry">
Davey Smith, G., and S. Ebrahim. 2003. <span>“<span class="nocase">’Mendelian randomization’: can genetic epidemiology contribute to understanding environmental determinants of disease?</span>”</span> <em>International Journal of Epidemiology</em> 32 (1): 1–22. <a href="https://doi.org/10.1093/ije/dyg070" class="external-link">https://doi.org/10.1093/ije/dyg070</a>.
</div>
<div id="ref-DaveySmithHemani2014" class="csl-entry">
Davey Smith, George, and Gibran Hemani. 2014. <span>“<span class="nocase">Mendelian randomization: genetic anchors for causal inference in epidemiological studies</span>.”</span> <em>Human Molecular Genetics</em> 23 (R1): R89—–R98. <a href="https://doi.org/10.1093/hmg/ddu328" class="external-link">https://doi.org/10.1093/hmg/ddu328</a>.
</div>
<div id="ref-Pierce2013" class="csl-entry">
Pierce, Brandon L, and Stephen Burgess. 2013. <span>“<span class="nocase">Efficient design for Mendelian randomization studies: subsample and 2-sample instrumental variable estimators.</span>”</span> <em>American Journal of Epidemiology</em> 178 (7): 1177–84. <a href="https://doi.org/10.1093/aje/kwt084" class="external-link">https://doi.org/10.1093/aje/kwt084</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
